src/components/flame-graph/FlameGraph.js

```diff
@@ -190,71 +190,81 @@ class FlameGraphImpl extends React.PureComponent<Props> {
         );
+        return;
       }
-      return;
-    }
 
-    switch (event.key) {
-      case 'ArrowDown': {
-        const prefix = callNodeTable.prefix[selectedCallNodeIndex];
-        if (prefix !== -1) {
-          changeSelectedCallNode(
-            threadsKey,
-            getCallNodePathFromIndex(prefix, callNodeTable)
-          );
+      switch (event.key) {
+        case 'ArrowDown': {
+          const prefix = callNodeTable.prefix[selectedCallNodeIndex];
+          if (prefix !== -1) {
+            changeSelectedCallNode(
+              threadsKey,
+              getCallNodePathFromIndex(prefix, callNodeTable)
+            );
+          }
+          break;
         }
-        break;
-      }
-      case 'ArrowUp': {
-        const [callNodeIndex] = callTree.getChildren(selectedCallNodeIndex);
-        // The call nodes returned from getChildren are sorted by
-        // total time in descending order.  The first one in the
-        // array, which is the one we pick, has the longest time and
-        // thus the widest box.
-
-        if (callNodeIndex !== undefined && this._wideEnough(callNodeIndex)) {
-          changeSelectedCallNode(
-            threadsKey,
-            getCallNodePathFromIndex(callNodeIndex, callNodeTable)
-          );
+        case 'ArrowUp': {
+          const [callNodeIndex] = callTree.getChildren(selectedCallNodeIndex);
+          // The call nodes returned from getChildren are sorted by
+          // total time in descending order.  The first one in the
+          // array, which is the one we pick, has the longest time and
+          // thus the widest box.
+
+          if (callNodeIndex !== undefined && this._wideEnough(callNodeIndex)) {
+            changeSelectedCallNode(
+              threadsKey,
+              getCallNodePathFromIndex(callNodeIndex, callNodeTable)
+            );
+          }
+          break;
         }
-        break;
-      }
-      case 'ArrowLeft':
-      case 'ArrowRight': {
-        const callNodeIndex = this._nextSelectableInRow(
-          selectedCallNodeIndex,
-          event.key === 'ArrowLeft' ? -1 : 1
-        );
-
-        if (callNodeIndex !== undefined) {
-          changeSelectedCallNode(
-            threadsKey,
-            getCallNodePathFromIndex(callNodeIndex, callNodeTable)
+        case 'ArrowLeft':
+        case 'ArrowRight': {
+          const callNodeIndex = this._nextSelectableInRow(
+            selectedCallNodeIndex,
+            event.key === 'ArrowLeft' ? -1 : 1
           );
+
+          if (callNodeIndex !== undefined) {
+            changeSelectedCallNode(
+              threadsKey,
+              getCallNodePathFromIndex(callNodeIndex, callNodeTable)
+            );
+          }
+          break;
         }
-        break;
+        default:
+          // We shouldn't arrive here, thanks to the if block at the top.
+          console.error(
+            `An unknown key "${event.key}" was pressed, this shouldn't happen.`
```

Comment:
Ah, interesting. This looks correct but it took me some time to understand the reason why we can't arrive to this path. Would you mind also adding a comment to `['ArrowDown', 'ArrowUp', 'ArrowLeft', 'ArrowRight'].includes(event.key)` if check to also mention to update this switch-case in case we change this array. (I know probably we are not going to change it frequently, but just in case)

Reply:
Good idea, thanks for the suggestion